src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -352,6 +375,55 @@ public async Task ProcessInputAsync(Message message, Flow flow, IContext message
             }
         }
 
+        private async Task ProcessGlobalOutputActions(IContext context, Flow flow, LazyInput lazyInput, InputTrace inputTrace, CancellationToken cancellationToken)
+        {
+            if (flow.OutputActions != null)
+            {
+                await ProcessActionsAsync(lazyInput, context, flow.OutputActions, inputTrace?.OutputActions, cancellationToken);
+            }
+        }
+
+        private async Task<(Flow, State)> RedirectToSubflowAsync(IContext context, Identity userIdentity, State state, Flow parentFlow, CancellationToken cancellationToken)
+        {
+            var shortNameOfSubflow = state.GetExtensionDataValue(SHORTNAME_OF_SUBFLOW_EXTENSION_DATA);
+            if (shortNameOfSubflow.IsNullOrEmpty())
+            {
+                throw new ArgumentNullException($"Error on redirect to subflow '{state.Id}'");
+            }
+
+            var subflow = await _flowLoader.LoadFlowAsync(FlowType.Subflow, parentFlow, shortNameOfSubflow, cancellationToken);
+            if (subflow == null)
+            {
+                throw new ArgumentNullException($"Error on return subflow '{shortNameOfSubflow}'");
+            }
+
+            subflow.Validate();
+            context.Flow = subflow;
+            var newState = subflow.States.Single(s => s.Root);
+
+            await _stateSessionManager.SetStateAsync(userIdentity, shortNameOfSubflow, cancellationToken);
+
+            return (subflow, newState);
+        }
+
+        private async Task<(Flow, State)> RedirectToParentFlowAsync(IContext context, Identity userIdentity, Flow flow, string parentStateId, CancellationToken cancellationToken)
+        {
+            if (context.Flow.Parent == null)
```

Comment:
You check if `context.Flow.Parent` is null but this value is never read.

Reply:
fixed